plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -556,12 +558,12 @@ data Context uni fun ann
     -- ^ @[V _]@
     | FrameApplyArg !(CekValEnv uni fun ann) !(NTerm uni fun ann) !(Context uni fun ann)
     -- ^ @[_ N]@
-    | FrameApplyValues ![CekValue uni fun ann] !(Context uni fun ann)
-    -- ^ @[_ V0...Vn]@
+    | FrameApplyArgVal !(CekValue uni fun ann) !(Context uni fun ann)
```

Comment:
I think we should benchmark this change separately. I _think_ I tried this before and it was worse. A priori it adds another traversal of the arguments that we don't need.

Reply:
> A priori it adds another traversal of the arguments that we don't need.

It's on par with using `DList`s, see [this](https://www.joachim-breitner.de/blog/620-Constructing_a_list_in_a_Monad) and [this](https://www.joachim-breitner.de/blog/684-Constructing_a_list_in_a_monad_revisited). Those `DList`s don't come for free!